locals.tf

```diff
@@ -1,6 +1,8 @@
 locals {
-  template                      = var.template == null ? [] : [var.template]
-  required_status_checks        = var.required_status_checks == null ? [] : [var.required_status_checks]
-  required_pull_request_reviews = var.required_pull_request_reviews == null ? [] : [var.required_pull_request_reviews]
-  restrictions                  = var.restrictions == null ? [] : [var.restrictions]
-}
\ No newline at end of file
+  template                         = var.template == null ? [] : [var.template]
+  required_status_checks           = var.required_status_checks == null ? [] : [var.required_status_checks]
+  required_pull_request_reviews    = var.required_pull_request_reviews == null ? [] : [var.required_pull_request_reviews]
+  required_pull_request_reviews_v3 = var.required_pull_request_reviews_v3 == null ? [] : [var.required_pull_request_reviews_v3]
+  restrictions                     = var.restrictions == null ? [] : [var.restrictions]
+  validate_protection_version      = var.use_branch_protection && var.use_branch_protection_v3 ? tobool("Sorry, you can only use either `github_branch_protection` or `github_branch_protection_v3`.") : true
```

Comment:
Hey @patrickmukumbu since there is no way to really validate two variables at the same time, that's why I am using a local to do 'something impossible' when both are set to true   (the error message guides the user on what to do though). Ideally, we only want to use either `branch_protection` or `branch_protection_v3`.

Reply:
@JohnNMwangi 
When you specify both from your end, does the error message come up?

Reply:
No, it throughs an error

Reply:
Yes I do, do you, the way you have highlighted it up there.

Reply:
Yeah, just the way I have highlighted up there

Reply:
Currently there is an open issue with HashiCorp, https://github.com/hashicorp/terraform/issues/25609, we can circle back later and improve the validation once a solution has been provided.